src/web/mjs/connectors/ManhwaEighteen.mjs

```diff
@@ -1,15 +1,59 @@
-import FlatManga from './templates/FlatManga.mjs';
+import Connector from '../engine/Connector.mjs';
 
-export default class ManhwaEighteen extends FlatManga {
+export default class ManhwaEighteen extends Connector {
 
     constructor() {
         super();
         super.id = 'manhwa18-int';
         super.label = 'Manhwa 18 (.net)';
         this.tags = [ 'hentai', 'multi-lingual' ];
         this.url = 'https://manhwa18.net';
-        this.requestOptions.headers.set('x-referer', this.url);
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            if (JSON.stringify(mangaList[mangaList.length - 1]) == undefined && mangas[mangas.length - 1] != undefined ) {
+                mangaList.push(...mangas);
+            } else if (JSON.stringify(mangaList[mangaList.length - 1].id) != JSON.stringify(mangas[mangas.length - 1].id)) {
+                if (mangas.length > 0) {
+                    mangaList.push(...mangas);
+                }
+            } else {
+                run = false;
+            }
+        }
```

Comment:
To remove that 502 Error, I think it's better to get total count of pages, because the site can return the same last page with increasing page parameter in the request.
```suggestion
        const uri = new URL('/manga-list.html?listType=pagination&sort=name&sort_type=ASC', this.url);
        const request = new Request(uri, this.requestOptions);
        const data = await this.fetchDOM(request, 'div.pagination-wrap ul.pagination li:nth-last-child(2) a');
        const pageCount = parseInt(data[0].text.trim());
        for (let page = 1; page <= pageCount; page++) {
            const mangas = await this._getMangasFromPage(page);
            mangaList.push(...mangas);
        }
```

Reply:
yours is a much cleaner solution. what i did was to compare the contents of the last push to the new ones to be pushed since anything beyond the last page is repeated with each request until like ~160 requests until it throws the 520 error.